src/react-table.d.ts

```diff
@@ -0,0 +1,29 @@
+﻿/* eslint-disable @typescript-eslint/no-unused-vars */
+import "@tanstack/react-table";
+import { RowData } from "@tanstack/react-table";
+import { CSSProperties } from "react";
+
+interface DropdownColumnFilterOption {
+  label: string;
+  value: string | number;
+}
+
+interface DropdownColumnFilter {
+  options: DropdownColumnFilterOption[];
+}
+
+export interface EnumValue {
+  value: number | string | undefined;
+  text: string;
+  disabled?: boolean;
+}
+
+declare module "@tanstack/table-core" {
+  interface ColumnMeta<TData extends RowData, TValue> {
+    dropdownFilter?: DropdownColumnFilter;
```

Comment:
@neotob 

I think something like `filterVariant` would be better? 
Possible options could be: `dropDown`, `textInput`, `numberInput` etc.

What do you think?

Reply:
enum values is removed, so we keep the dropdownFilter.options.